lib/matplotlib/__init__.py

```diff
@@ -562,7 +552,17 @@ def _get_config_or_cache_dir(xdg_base):
     else:
         if os.access(str(configdir), os.W_OK) and configdir.is_dir():
             return str(configdir)
-    return _create_tmp_config_or_cache_dir()
+    # If the config or cache directory cannot be created or is not a writable
+    # directory, create a temporary one.
+    tmpdir = os.environ["MPLCONFIGDIR"] = \
+        tempfile.mkdtemp(prefix="matplotlib-")
+    atexit.register(shutil.rmtree, tmpdir)
+    _log.warning(
+        "Matplotlib created a temporary config/cache directory at %s because "
+        "the default path (%s) is not a writable directory; it is recommended "
```

Comment:
To speed up matplotlib import it is recommended...

Reply:
actually that's not the only reason; safety against fork()ing is the other one (i.e. #15911).  OTOH that's a pretty obscure case and I'm not sure mentioning it in the warning is really helpful.

Reply:
It would be good give a rough motivation for the recommendation (What's the benefit for doing/what's the downside of not doing?). I don't think it has to be exhaustive, so leaving out rare additional effects (fork safety) is ok.

Reply:
ok, done

Reply:
Slightly reworded to explicitly mention multiprocessing, as that's also what happened in https://mail.python.org/pipermail/matplotlib-users/2020-January/001820.html (I think).